src/components/offcanvas/__tests__/Offcanvas.test.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import {render} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Offcanvas from '../Offcanvas';
+
+jest.useFakeTimers();
+
+describe('Offcanvas', () => {
+  test('renders a div with class "offcanvas"', () => {
+    render(<Offcanvas is_open />);
+
+    expect(document.body.querySelector('div.offcanvas')).not.toBe(null);
+  });
+
+  test('toggle visibility with "is_open"', () => {
+    const {rerender} = render(<Offcanvas />);
+
+    expect(document.body.querySelector('.offcanvas')).toBe(null);
+
+    rerender(<Offcanvas is_open />);
+    jest.runAllTimers();
+
+    expect(document.body.querySelector('.offcanvas')).not.toBe(null);
+
+    rerender(<Offcanvas />);
+    jest.runAllTimers();
+
+    expect(document.body.querySelector('.offcanvas')).toBe(null);
+  });
+
+  test('renders its content', () => {
+    render(<Offcanvas is_open>Some offcanvas content</Offcanvas>);
+    expect(document.body.querySelector('.offcanvas')).toHaveTextContent(
+      'Some offcanvas content'
+    );
+  });
+
+  test('applies additional CSS classes with props', () => {
+    // placement offcanvas
+    const placements = ['end', 'top', 'bottom', 'start'];
+    for (let i = 0; i < placements.length; i++) {
+      rerender(<Offcanvas is_open placement={placements[i]} />);
+      expect(document.body.querySelector('.offcanvas-dialog')).toHaveClass(
+        `offcanvas-${placements[i]}`
+      );
+    }
+
+    // scrollable content
+    const {rerender} = render(<Offcanvas is_open />);
+    // TODO: Check body style="overflow: hidden;" when scrollable=false (default)
```

Comment:
you can do `.toHaveStyle({overflow: "hidden"})` for this

Reply:
Would it need to be

```js
expect(document.body.querySelector('.body')).toHaveStyle({
      overflow: 'hidden'
    });
```

or is the `.querySelector` not needed?

```js
expect(document.body).toHaveStyle({overflow: 'hidden'});
```

Reply:
Frustratingly due to the way the tests work, it's not applying the style correctly to the body, so I'm unable to run this test. I've removed it and if required can be revisited at a later date.